CustomerAPI/Controllers/CustomerController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Data.Entities;
+using Data.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CustomerAPI.Controllers
+{
+    [Route("api/controller")]
+    [ApiController]
+    public class CustomerController : ControllerBase
+    {
+        private ICustomerService _service;
+
+        public CustomerController(ICustomerService service)
+        {
+            _service = service ?? throw new ArgumentNullException(nameof(service));
+        }
+
+        [HttpPost]
+        public IActionResult Create(Customer customer)
+        {
+            try
+            {
+                _service.Create(customer);
+                return Created("Id: ", customer.Id);
+            }
+            catch (ArgumentException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+        }
+
+        [HttpGet("{Id}")]
+        public IActionResult GetById(long Id)
+        {
+            try
+            {
+                var customer = _service.GetById(Id);
+                return Ok(customer);
+            }
+            catch (ArgumentNullException exception)
+            {
+                return NotFound(exception.Message);
+            }
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            try
+            {
+                var listCustomers = _service.GetAll();
+                return Ok(listCustomers);
+            }
+            catch
+            {
+                return NoContent();
+            }
```

Comment:
Apenas devemos retornar 204 quando o endpoint não possui retorno o que nunca vai ser o caso aqui, como estamos fazendo GetAll ou vai retornar os dados ou uma lista vazia quando não tiver nenhum dado, ou seja, possui retorno. CC: @gregoryzimmerwarren 
```suggestion
            try
            {
                var listCustomers = _service.GetAll();
                return Ok(listCustomers);
            }
            catch (Exception ex)
            {
                var exMessage = ex.InnerException?.Message ?? ex.Message;
                return Problem(exMessage);
            }
```